### PR TITLE
docs: update API docs with async

### PR DESCRIPTION
**Pull request by lzpap**
### Details
Add async API classes and their coroutines to documentation.

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -41,19 +46,48 @@ implementation point of view, :py:class:`Iota` is a subclass of
...
-Take a look on the class definitions and notice that :py:class:`Iota`
-has a :py:class:`Seed` attribute. This is becasue the Extended API is able
-to generate private keys, addresses and signatures from your seed.
-**Your seed never leaves the library and your machine!**
+To use the functionally same, but asynchronous API classes, you can do the
+following:
+
+.. code-block::
+   :linenos:
+   :emphasize-lines: 3,4
```
Comment (todofixthis) [inline docs/api.rst]: 👍 Ooh, nice!

File: docs/multisig.rst [hunk 2]
```diff
@@ -1,24 +1,98 @@
...
-It is standard functionality in blockchain systems and it is also implemented in IOTA
+It is standard functionality in blockchain systems and it is also implemented
+in IOTA.
 
 .. note::
 
     You can read more about IOTA multisignature on the `wiki`_.
 
+First, we will take a look on what `Multisig API`_ (s) you can use, and what
+`PyOTA Multisignature Types`_ are there fore you if the standard API is not
```
Comment (todofixthis) [inline docs/multisig.rst]:
```suggestion
`PyOTA Multisignature Types`_ are there for you if the standard API is not
```

File: iota/multisig/api.py [hunk 3]
```diff
@@ -188,20 +206,20 @@ class AsyncMultisigIota(AsyncIota):
...
 
             If the bundle has no unspent inputs, ``change_address` is
             ignored.
 
             .. important::
-                Unlike :py:meth:`iota.api.Iota.prepare_transfer`, this
+                Unlike :py:meth:`~iota.Iota.prepare_transfer`, this
                 method will NOT generate a change address automatically.
-                If there are unspent inputs and ``change_address`` is
+
```
Comment (todofixthis) [inline iota/multisig/api.py]: ❔ Was this line meant to be removed?

File: iota/multisig/api.py [hunk 4]
```diff
@@ -415,20 +451,20 @@ def prepare_multisig_transfer(
...
```
Comment (todofixthis) [inline iota/multisig/api.py]: ❔ Also here

File: iota/multisig/types.py [hunk 5]
```diff
@@ -21,6 +21,29 @@ class MultisigAddress(Address):
     In order to spend inputs from a multisig address, the same private
     keys must be used, in the same order that the corresponding digests
     were used to generate the address.
+
+    .. note::
+        Usually, you don't have to create a MultisigAddress manually. Use
+        :py:meth:`~iota.multisig.MultisigIota.create_multisig_address` to
+        derive and address from a list of digests.
```
Comment (todofixthis) [inline iota/multisig/types.py]:
```suggestion
        derive an address from a list of digests.
```

File: iota/multisig/api.py (hunk 3 above)
Comment (lzpap) [reply iota/multisig/api.py]: good catch, it wasn't

File: iota/multisig/api.py (hunk 4 above)
Comment (lzpap) [reply iota/multisig/api.py]: adding it back.

**Pull request merged**